pwndbg/gdblib/events.py

```diff
@@ -97,7 +100,9 @@ def on_stop(self) -> None:
 objfile_cache: Dict[str, Set[str]] = {}
 
 
-def connect(func: Callable[P, T], event_handler: Any, name: str = "") -> Callable[P, T]:
+def connect(
+    func: Callable[P, T], event_handler: Any, name: str = "", priority: bool = False
```

Comment:
@gsingh93 thoughts on this? I believe we need it so that clearing cache is triggered as very first event, but I'm not sure I like a boolean value for it. I mean, this eventually means calling multiple 'priority' events will end up with the same issue as we had before.

Reply:
Yea, `priority` is usually an enum or an int.

Also, my general thought about adding priority to these events is that we need to really think about the right way to do this, otherwise this is going to turn into a mess and get completely rewritten at some point when it becomes hard to maintain, or doesn't solve the problem well in the long run. And I don't have the time to think through it. But I guess if it's fixing an actual bug and we don't have an alternative solution it makes sense to merge it? Up to you.

Reply:
I can switch it to an enum priority. Are there any other concerns about the implementation?

Reply:
Let's merge this as of now. I don't think we need to make it more complex with int/enum for now :).

Reply:
Oh, we already made a switch. I guess that is fine.